.github/actions/static-code-analysis/action.yml

```diff
@@ -0,0 +1,29 @@
+name: PowerShell Static Code Analysis
+description: Runs static code analysis on PowerShell scripts and cmdlets.
+
+inputs:
+  run-windows-powershell:
+    required: false
+    default: "true"
+    description: >
+      If set to "true", static code analysis is ran through Windows PowerShell (version 5.x), but only on Windows.
+      When not "true", machine-types shouldn't have any Windows-based entries, because they won't do anything.
```

Comment:
This line confuses me.
1. By "entries" did you mean "runners" or something else?
2. If this is not "true" but "run-powershell-core" is, then the analysis will still execute on `pwsh` regardless whether the runner is Windows or Linux, so that's not nothing.

Same below in the workflow.

Reply:
1. Yes, "runner" is the correct term here (but by entry I meant that an element of the array that defines a Windows-based runner).
2. I managed to confuse myself! What I wanted to write is kind of the opposite: When `run-powershell-core` is not true, then Linux-based runners can be omitted, because those won't run Windows PS-based analysis, so they won't do anything in the end. I'll update the text.